index.js

```diff
@@ -52,12 +56,21 @@ function keyPair (storage, derive) {
     function createNew () {
       derive(null, (err, res) => {
         if (err) return cb(err)
-        writeName(res.name, err => {
```

Comment:
i have a tingling sensation that the order here was important, to avoid corruption

Reply:
but i'm unsure :P

Reply:
know why that'd be? if the key write fails the whole thing will be re-created on the next load.

Reply:
yea i think you're right